Created Service policies for Google BeyondCorp

s223880826 opened the pull request:
I wrote 18 different policies for the BeyondCorp service.

christoraju085 committed:
Initial Commit

visal-dam committed:
Update get_resource_name (#104)

Updated get_resource_name to handle when there's two 'name' values. Thank you to Sebastian Edge for pointing this out and for providing this fix.

JSchwab100 committed:
Documentation/initialstructure (#122)

* Add policy and examples for Private CA certificate lifetime

Added documentation, Terraform examples, and OPA policy scaffolding for managing the lifetime of google_privateca_certificate_authority resources in GCP. This includes example configs, a policy template for lifetime enforcement, and supporting variable definitions. Also removed obsolete VSCode settings.

* Pushing documentation

* Add initial content to index.md

* Add relative link to API Gateway documentation

* Convert index.md to index.html with fetch script

* Refactor script to dynamically fetch and display files

* Update docs index.html to fetch GitHub files

Refactored script to fetch and display files from GitHub repository directly in the HTML body.

* Refactor HTML script to fetch repo contents correctly

* Update API endpoint in index.html

* Update fetch URL in index.html

* Update API endpoint in index.html

* Update title and heading for GCP documentation

* Refactor script to improve folder rendering logic

* Refactor HTML and JavaScript for GCP documentation

* Add initial HTML structure for documentation

* Update file links to include Policy-Deployment-Engine

* Fix duplicate data fetching in index.html

* Fix suburl construction for GitHub API fetch

* Limit displayed items to 2 at each level

* Refactor file fetching and limit display items

* Modify display name and path handling for markdown files

* Enhance index.html with styling and search feature

* Update file fetching and display logic in index.html

* Update index.html

* Display all top-level folders instead of two

Remove limit on top-level folder display

* Change title and enhance search functionality

Updated the title and added a warning message about API call limits. Adjusted styling and search functionality for better usability.

* Moving script

* New document generator

* Rearranged script

* Restore ReadME

* Removing unnecessary index.html

* Providing instructions for doc generation

Deleted generated AlloyDB markdown documentation files and updated the resource_json for alloydb_backup to remove security/compliance metadata. Added a README to scripts explaining the documentation workflow, renamed markdown_builder.py to create_markdown.py, and doc_generator.py to original_document_generator.py for clarity.

* Create subfolder for docgen scripts and move README

JSchwab100 committed:
Fix/json compiling (#125)

* Normalize security_impact as boolean in docgen

Updated the docgen README and create_markdown.py to treat 'security_impact' as a boolean value instead of a string or null. Adjusted normalization logic and documentation to clarify expected types for 'required' and 'security_impact' fields.

* Update docgen README with vertex_ai example

Replaces references to 'alloydb' with 'vertex_ai' in the documentation and updates the example JSON resource to use 'vertex_ai_dataset' with more detailed argument documentation. This provides clearer guidance for documenting Vertex AI services.

sumedh004 commented on policies/gcp/beyondcorp/google_beyondcorp_app_connection/gateway_whitelist/vars.rego:
There should be only one vars.rego file per resource folder. Remove it from every policy folder. Do this for all of the resources

s223880826 replied:
Done

sumedh004 commented on inputs/gcp/beyondcorp/google_beyondcorp_app_connection/gateway_whitelist/c.tf:
```diff
@@ -0,0 +1,18 @@
+resource "google_beyondcorp_app_gateway" "c" {
+  name = "my-app-gateway"
```
resource naming "c" should be only for the resource that is being tested. In this case you are checking for google_beyondcorp_app_connection so that is correct. However it should be different for google_beyondcorp_app_gateway and referenced correctly. To be done for all tf files

s223880826 replied:
Done

sumedh004 commented on policies/gcp/beyondcorp/google_beyondcorp_app_connection/gateway_whitelist/vars.rego:
```diff
@@ -0,0 +1,8 @@
+variables := {
+  "friendly_resource_name": "BeyondCorp AppConnection Gateway",
+  "resource_type": "google_beyondcorp_app_connection",
+  "resource_value_name": "gateway.app_gateway"
```
Wrong value name.also referencing is incorrect. should only reference the attribute from the resource which is google_beyondcorp_app_connection in this case. You have used attribute from another resource gateway. Applicable for all vars.rego files

s223880826 replied:
Changed and Done

sumedh004 commented on inputs/gcp/beyondcorp/google_beyondcorp_app_connection/region_whitelist/c.tf:
```diff
@@ -0,0 +1,10 @@
+  name = "c"
+  type = "TCP_PROXY"
+  region = "australia-southeast1"
+  display_name = "region good"
```
Keep minimal attributes that are required to create the tf. Optional attributes such as display_name in this case should be removed. Applicable for all tf files

s223880826 replied:
Changed

sumedh004 commented on inputs/gcp/beyondcorp/google_beyondcorp_app_connection/port_whitelist/config.tf:
```diff
@@ -0,0 +1,15 @@
+}
+
+provider "google" {
+  project = "smooth-verve-467716-v1"
```
Do not edit config.tf file. If project is required include that field in the tf files. For location as well. This is to be done for all tf files

s223880826 replied:
Sure...Put the normal config

sumedh004 commented on policies/gcp/beyondcorp/google_beyondcorp_app_connection/gateway_whitelist/policy.rego:
Not security related. Remove this policy

s223880826 replied:
Removed

sumedh004 commented on policies/gcp/beyondcorp/google_beyondcorp_app_connection/port_whitelist/policy.rego:
In this you can add another situation which checks for IP addresses of host

s223880826 replied:
Done

sumedh004 replied:
There should be one more resource nc1 in tf to check situation 2 as currently its not flagging nc1

<img width="1061" height="196" alt="image" src="https://github.com/user-attachments/assets/46303de1-995c-4ca9-a7e5-1965e9606425" />

sumedh004 commented on policies/gcp/beyondcorp/google_beyondcorp_app_connection/region_whitelist/policy.rego:
```diff
@@ -0,0 +1,24 @@
+    {
+      "condition": "Region must be inside the approved one",
+      "attribute_path": ["region"],
+      "values": ["australia-southeast1"],
```
check if australia-southeast1 is the only possible location in Aus. If there are multiple, then those Aus locations should be included as well in the list

s223880826 replied:
Done

sumedh004 commented on policies/gcp/beyondcorp/google_beyondcorp_app_connection/type_whitelist/policy.rego:
No need for this policy as this will be the default one used if unspecified

s223880826 replied:
Done

sumedh004 commented on policies/gcp/beyondcorp/google_beyondcorp_app_connector/region_whitelist/policy.rego:
```diff
@@ -0,0 +1,24 @@
+    {
+      "condition": "Region must be inside the approved one",
+      "attribute_path": ["region"],
+      "values": ["australia-southeast1"],
```
check if australia-southeast1 is the only possible location in Aus. If there are multiple, then those Aus locations should be included as well in the list

s223880826 replied:
Done

sumedh004 commented on policies/gcp/beyondcorp/google_beyondcorp_app_gateway/host_type_whitelist/policy.rego:
```diff
@@ -0,0 +1,23 @@
+    {
+      "condition": "Host type must be GCP_REGIONAL_MIG.",
+      "attribute_path": ["host_type"],
+      "values": ["GCP_REGIONAL_MIG"],
```
Any particular reason for this being security related?

s223880826 replied:
I think that host type is required to enter to that particular gateway.

sumedh004 commented on policies/gcp/beyondcorp/google_beyondcorp_app_gateway/region_whitelist/policy.rego:
```diff
@@ -0,0 +1,23 @@
+    {
+      "condition": "Region must be whitelisted",
+      "attribute_path": ["region"],
+      "values": ["australia-southeast1"],
```
check if australia-southeast1 is the only possible location in Aus. If there are multiple, then those Aus locations should be included as well in the list

s223880826 replied:
Done

sumedh004 commented on policies/gcp/beyondcorp/google_beyondcorp_app_gateway/type_whitelist/policy.rego:
```diff
@@ -0,0 +1,23 @@
+    {
+      "condition": "Type must be TCP_PROXY",
+      "attribute_path": ["type"],
+      "values": ["TCP_PROXY"],
```
No need for this policy as this will be the default one used if unspecified

s223880826 replied:
Done

sumedh004 commented on policies/gcp/beyondcorp/google_beyondcorp_security_gateway/hubs_region_whitelist/policy.rego:
```diff
@@ -0,0 +1,23 @@
+    {
+      "condition": "Hubs region must be 'australia-southeast1'",
+      "attribute_path": ["hubs", 0, "region"],
+      "values": ["australia-southeast1"],
```
Check locations as mentioned previously

s223880826 replied:
Done

sumedh004 commented on policies/gcp/beyondcorp/google_beyondcorp_security_gateway/security_gateway_id_pattern/policy.rego:
Remove this policy as checking names is not security related

s223880826 replied:
Done

sumedh004 replied:
You havent removed this policy. Its still there

sumedh004 commented on policies/gcp/beyondcorp/google_beyondcorp_security_gateway_application/application_id_pattern/policy.rego:
Not security related. Remove this policy

s223880826 replied:
Removed

sumedh004 commented on policies/gcp/beyondcorp/google_beyondcorp_security_gateway_application/endpoint_hostname_whitelist/policy.rego:
```diff
@@ -0,0 +1,35 @@
+  #   {
+  #     "situation_description": "Endpoint hostname is outside approved domains.",
+  #     "remedies": ["Use hostnames ending with '.com' only."]
+  #   },
```
Remove the commented code

s223880826 replied:
Done

sumedh004 commented on policies/gcp/beyondcorp/google_beyondcorp_security_gateway_application/upstreams_whitelist/policy.rego:
```diff
@@ -0,0 +1,38 @@
+    {
+      "condition": "Upstreams egress regions must be in the approved list.",
+      "attribute_path": ["upstreams", 0 , "egress_policy", 0 , "regions"],
+      "values": ["australia-southeast1"],
```
check for locations as mentioned previously

s223880826 replied:
Done

sumedh004 commented on policies/gcp/beyondcorp/google_beyondcorp_security_gateway_application/upstreams_whitelist/policy.rego:
```diff
@@ -0,0 +1,38 @@
+        "projects/my-proj/global/networks/prod-vpc",
+        "projects/my-proj/global/networks/shared-services-vpc"
+      ],
+      "policy_type": "whitelist"
```
Pattern whitelist better suited

s223880826 replied:
Done

sumedh004 commented on inputs/gcp/beyondcorp/google_beyondcorp_security_gateway_application/endpoint_hostname_whitelist/c.tf:
```diff
@@ -0,0 +1,8 @@
+resource "google_beyondcorp_security_gateway_application" "c" {
+  security_gateway_id = "c-default-sg"
```
For whatever is used in your vars.rego as your "resource_value_name" needs to be in our standard format as this is what will be output when running your policies. We want policies to output a standard format so we want whatever attribute you have put in the vars.rego "resource_value_name" to be of the format "c" or "nc" with possible digits after if there are multiple resource instances.

s223880826 replied:
Sure..Done

sumedh004 reviewed (changes requested):
There should be only one vars.rego file per resource folder. Remove it from every policy folder. Do this for all of the resources. Since you have vars.rego in every policy folder the opa eval command is passing all of c as well as nc and I cannot check if policies are working as intended.



Also in the meantime do this for all policies:
In the vars.rego file, whatever is declared as resource_value_name, should be nameed with either c or nc in the corresponding Terraform files.

For example, if resource_value_name is set to rule, the value in the Terraform file should be c/nc as:
rule = "c" and so on for c.tf
rule = "nc" and so on for nc.tf

This applies to all of your resources.

Varniah88 committed:
Security Policies Service - Model Armor (#118)

* Security Policies Service - Model Armor

* policies fix

* Adding documentation

Varniah88 committed:
Cloud Functions Gen 2 - Security Policies (#83)

* Policy Create - service:cloud function gen2

* small revert

* small revert

* clean code

* Security Policies

* code clean

* code clean

* code clean

* naming changes

* naming changes

* Non security related policy removal and location update

* Non security related policy removal and location update

* policies fix

* Adding Documentation

KeerthanaVijekumar committed:
Identity_Aware_Proxy code import (#60)

* Identity_Aware_Proxy code import

* identity_aware_proxy service's google_iap_brand resource changes

* identity_aware_proxy changes and new attributes

* IAP Changes

* IAp Fixes

* IAP Resource policy update

* google_iap_setting.allowed_domain condition fixed

* Identity_Aware_Proxy Documentation

MenuraDewalegama committed:
adding documentation (#158)

jashan-lefty committed:
Linters script (#160)

* first commit

* update the inputs folder logic

Seb-sketch committed:
Update helpers.rego (#166)

Pat-Stuart committed:
Revert "Update helpers.rego (#166)" (#167)

This reverts commit 6af54154b58408d2a7c72b95691081f435c17ce8.

sumedh004 committed:
Gcp/service/cloud storage (#54)

* gcp cloud service storage

* updated asked changes

* asked changes by yotam

* naming cleanup

* naming cleanup

* default object access control resource

* policies for some resources

* add remaining policies

* reviewed existing inherited policies

* minor format

* address pr comments

* combine some policies

* naming cleanup

* various bug fixes and renaming for testing

* merge commit

* folders rename

* terraform fmt

* add documentation

* folder rename and documentation added

* doc generated

* slight update to docs

* slight update to docs

---------

Co-authored-by: AkashreddyVootk <s224127166@deakin.edu.au>

Vamsi57366 committed:
Finished contributing for Application Integration (#105)

* Restored deleted folders and applied review feedback

* Addressed feedback and added documentation

vinvincodes committed:
Gcp/service/cloud platform (#106)

* Work on cloud platform service

* add inputs & policies of cloud_platform

* Delete .vscode/settings.json

* delete template comments from all policy files

* undo changes to templates

* delete comput and storage folders

* update

* fix: applied contributor feedback

* Save local changes before merging

* update to name for  resource_value_name on vars.rego

* remove exempt_members policy due to doesn't work and helper.rego hasn't update for this policy

* Add/update JSON documentation for GCP resources - cloud platform

* Add/update markdown documentation for GCP resources - cloud platform

* all updated

* readme.md back

* Fix: resolved merge conflict in cloudfunctions2_function.json, removed markers and applied origin/dev security rules

* Fix: resolved merge conflict in cloudfunctions2_function_iam.json, removed markers and applied origin/dev security rules

* Fix: resolved merge conflict in model_armor_template.json & _floorsetting.json, removed markers and applied origin/dev security rules

* google_project update name

* update

* update google_service_account_key/exposure

* update google_folder_organization_policy/constraint

AK199417 committed:
Gcp/service/google cloud netapp volumes (#74)

Pat-Stuart committed:
Testing Script and CI Pipeline (#117)

* finished script and basic pipeline

* remove comment

* finished script and basic pipeline

* remove comment

jashan-lefty committed:
linter minor update (#173)

* first commit

* update the inputs folder logic

* Fix: Root path error fix (Alloes to run linters from any path) but it must me called from the root directory "Policy-Deployment-Engine

jashan-lefty committed:
Cloud key management CleanUP from T1 (#164)

* Add policies and inputs for Cloud Key Management

* Updated helpers

* updating

* Updating the policies

* final updates

* Changes after the review

* Initial changes

* initial changs

* vars repositioning

* Changes in naming

* final changes into, updating according to linters and pipeline

* final commits

* updated gitignore

* added readme

* update vars

Pat-Stuart committed:
clean up (#151)

sennan21 committed:
feat(looker-core): add 11 Looker Core security policies with documentation (#172)

- Added 11 Looker Core security policies with proper OPA rules
- Fixed duplicate variable declarations by creating separate variables.tf files
- Included comprehensive documentation with security annotations
- All policies pass terraform init/validate without errors
- Using proper service branch structure as requested

sumedh004 committed:
restore policy template (#182)

Good catch sumedh

Miguwell-I committed:
Created: gcp/service/privileged access manager (#112)

* Initial Commit

* Updated privileged access manager entitlement policies

* added changes

* small change

* remove policy

sumedh004 commented:
have you pushed the changes?

sumedh004 committed:
Feature/folder generator (#183)

* removed selenium scrapping

* remove cache json for service to use runtime cache

* create readme

* create readme

shivam3175 committed:
Finished policies on Google Lustre service (#114)

* Initial commit

* Fixed review comments

* Add all plan.json files

* Pushing Lustre branch

* Updated few files

* Added documentation

* Updated changes

* Updated

* Removed chronicle inputs and policy folder

Sundi202 committed:
GKE Hub Policies  (#107)

* GKE Hub Feature (Fleet Observability) -  require default logs routing mode COPY or MOVE (updated)

* Security policies for google_gke_hub_feature

* no_public_principals_binding policy for IAM binding

* no_public_principals_member policy for IAM member

* configmanagement security policies (git_secure_auth, git_approved_HTTPS, git_approved_dir, pc_enable_required)

* added policy_binding policy & authority_issuer policy

* GKEHub: remove deprecated google_gke_hub_feature_membership/git_approved_dir

* GKEHub: standardize resource_value_name to {c|nc}; fix authority_issuer; tighten RBAC/HTTPS controls

shivam3175 committed:
Finished 10 Policies on Chronicle Service (#82)

s223880826 commented:
> have you pushed the changes?

No..i'm having some issues with push..Trying to solve it.

christoraju085 committed:
Accomodated the changes

s223880826 commented:
> have you pushed the changes?

Just now i pushed the changes. Do i need to pull that again?

christoraju085 committed:
Accomodated the changes

sumedh004 commented on inputs/gcp/beyondcorp/google_beyondcorp_app_connection/port_whitelist/c.tf:
```diff
@@ -1,5 +1,10 @@
+data "google_project" "project" {
```
Why did you added this data block now? We dont want that as the tf commands wont run. Instead use any name in project field and that still will work.

Do this for all tf files

sumedh004 commented on inputs/gcp/beyondcorp/google_beyondcorp_app_connection/port_whitelist/c.tf:
```diff
@@ -1,5 +1,10 @@
 resource "google_beyondcorp_app_connection" "c" {
-  name         = "c-port-ok"
+  name         = "c"
+  project      = data.google_project.project.id
```
here you can have directly project="whatever project name you want to enter"

sumedh004 commented on inputs/gcp/beyondcorp/google_beyondcorp_app_connection/port_whitelist/nc.tf:
```diff
@@ -1,5 +1,10 @@
 resource "google_beyondcorp_app_connection" "nc" {
-  name         = "nc-port-bad"
+  name         = "nc"
+  project      = data.google_project.project_nc.id
```
need to change referencing acordingly for all tf files. instead of referring to data source keep project ="any name you want"

Seb-sketch committed:
Revert "feat(looker-core): add 11 Looker Core security policies with document…" (#189)

This reverts commit d1b10d5e8381f4d68a3420d40deffe39b80ddf73.

Seb-sketch committed:
Gcp/service/cloud_deploy (Old PR from forked repo not cloned branch) (#186)

* GKE Hub Policies  (#107)

* GKE Hub Feature (Fleet Observability) -  require default logs routing mode COPY or MOVE (updated)

* Security policies for google_gke_hub_feature

* no_public_principals_binding policy for IAM binding

* no_public_principals_member policy for IAM member

* configmanagement security policies (git_secure_auth, git_approved_HTTPS, git_approved_dir, pc_enable_required)

* added policy_binding policy & authority_issuer policy

* GKEHub: remove deprecated google_gke_hub_feature_membership/git_approved_dir

* GKEHub: standardize resource_value_name to {c|nc}; fix authority_issuer; tighten RBAC/HTTPS controls

* Policies and documentation

* helpers comment

* Policies and terraform files from older branch

* Required changes to helpers needed for policies

* Fix wrong edit

* Fixing naming for input files to be exact matches "c" or "nc"

* Missed name from last commit, also formatting plan files

* Changing name of folders to more descriptive label

---------

Co-authored-by: Sebastian <sebastianedge50@gmail.com>

HarshPandya7 committed:
GDCE Service Completed  (#98)

* Updated GDCE policies and input files for cluster security compliance

* Updated GDCE policies, inputs, and documentation; added Cloud Stackdriver Logging service

* "Updated Document of GDCE"

* feat: add GDCE plan.json outputs only

* Fixed GDCE policies: corrected node pool locations, KMS key validation, cluster target version, and maintenance policy checks. Removed CloudStackdriver logging service.

* Remove Cloud Stackdriver Logging service from inputs and policies

* added GDCE cluster CIDR blocks policy

patrickandersondeakin committed:
API Hub pull request (#93)

* initial policy for dlp

* temporary helpers.rego fix

* polciies for network resource

* polciies for network resource

* policies for network peering resource

* policies for network policy resource

* policies for external access rule

* merged helper rego

* policy rewrite for new helper

* fix policy

* policy for external access rule

* remove project from config

* restructure and minor changes

* folder cleanup

* Add API Hub policies and inputs for peer review.

* Checking policies and code

* API Hub pull request

* Changes made according to Pat Stewart PR

* Added some more changes according to Pat Stuart PR

* Markdown created for peer review

* Peer review changes made for Api_Hub

* Made two minor changes

---------

Co-authored-by: Sumedh Vartak <67580847+sumedh004@users.noreply.github.com>

LegendaryMercury committed:
Discovery engine PR Request (#81)

* Fixed Mistakes

* Made Some Compliancy Changes

* Made Some Requested Changes

Made some Changes

* Added Plan Files

Added Plan Files

* Pattern_whitelist

Changed 2 policys to pattern whitelist

* Changed_engine_schema_json

PatternWhitelist

* Updated_recommendation_engine

Removed unneeded stuff.

* Taking out Trash

Gone
(the location policys that arent in AUS)
(also file adjustments for the engine_schema json check, back to whitelist but with full expression)

* More Changes

More Changes Made

* Added location Policy's Back

Yotam Said i could add them, because i did a few they will just be worth less.

* misc

some misc changes.

* Documentation

Hope this is what you wanted.

* search_engine_fix

* engine_config_fix

* Opps

Opps

---------

Co-authored-by: Pat Stuart [SSW] <71543754+Pat-Stuart@users.noreply.github.com>

sumedh004 commented on inputs/gcp/beyondcorp/google_beyondcorp_app_connector/service_account_whitelist/c.tf:
```diff
@@ -1,14 +1,19 @@
   principal_info {
     service_account { 
-      email = google_service_account.c.email 
+      email = google_service_account.accnt_c.email 
```
you havent put email in service account resource. this is causing  c to be flagged as non compliant in the policy

sumedh004 reviewed (changes requested):
Need to remove data block from all tf resources as mentioned in one comment for tf to work.

Need to check all resources which have public_access_prevention policies as none of them are working as intended

Make sure to check if policies are working correctly before pushing the code

<img width="1071" height="166" alt="image" src="https://github.com/user-attachments/assets/9ce6dee0-b013-4dda-b009-faa4f38ae4ab" />

Chhunly-TAING committed:
Gcp/service/bigquery data transfer (#121)

* Added BigQuery Data transfer policies

* implement BigQuery Data Transfer policy

* I have made some changes to my c.tf and nc.tf

* Address the feedback

* Push with the documentation

* I have fixed the output of the opa eval

* I have deleted sensitive_params

* I am sorry Sumedh, I forgotten to delete the document fro mthe GitHub

Prem-CyberSec committed:
Gcp/service/os config v2 (#195)

* Restarted the branch due to conflicts with files for other services

* Added a documentation in the README for understanding the first policy deployment

* Added policy for the new resource Policy_Orchestrator_for_folder

This policy works mostly the same as the one seen in other resources, this is covering the attribute orchestrated_resource

* Added another policy for policy_orchestrator_for_folder

ACTION policy is created for the policy_orchestrator_for_folder

* Added a new policy for policy_orchestrator_for_folder

Policy added for orchestration scope which specifies the location

* Reverted README.md to original file matching from DEV

* Added Documentations required for the service

* PATCH FOR THE ORIGINAL FILES

* modified:   inputs/gcp/os_config_v2/policy_orchestrator/action/nc.tf

---------

Co-authored-by: Seb-sketch <SebastianE50@pm.me>

visal-dam committed:
Created helpers.md (#196)

Technical documentation for the helpers.rego for future developers and maintainers.

MenuraDewalegama committed:
Gcp/service/bigquery data policy (#63)

* adding bigquery_data policies

* fixing indentation issues

* removing NSR policies

* fixing the vars.rego file name

* whitelisting all the Australian zones

* chnaging the value of resource_value_name

* expanding the policy to blacklist against allUsers and other conditions

* adding documentation

waffa-alhasani committed:
completed gcp Firebase service (#76)

* completed gcp Firebase service

* completed gcp Firebase service

sennan21 committed:
fix(looker_core) rename input resource labels to c/nc (#193)

* fix(looker_core): rename input resource labels to c/nc; regenerate lockfiles

* fix(looker_core): change name attribute from ok/bad to c/nc in c.tf & nc.tf (per PR #193 review)

* fix(looker_core): update name attributes to cc/nc and regenerate plan.json files (per PR #193 review)

* fix(looker_core): update name attributes to c1/nc per vars.rego specification and regenerate plan.json files

* fix(looker_core): set name values to c1/nc1 per reviewer guidance

* fix(looker_core): ensure c1/nc1 naming is consistent across all policies (per PR #193 review)

s223880826 committed:
Delete policies/gcp/beyondcorp/google_beyondcorp_security_gateway/security_gateway_id_pattern directory

sennan21 committed:
Gcp/service/recaptcha enterprise docs (#192)

* feat(gcp/service/recaptcha_enterprise): add allow_all_domains policy + vars + c/nc + config.tf (config unchanged)

* chore(recaptcha_enterprise): move service-level vars.rego to key/

* GCP reCAPTCHA Enterprise key: add policies for allow_amp_traffic, challenge_security_preference, integration_type (inputs + policy)

* docs(recaptchaenterprise): annotate web_settings security fields and generate markdown

* recaptchaenterprise/key: remove duplicate vars; add test evidence (.terraform.lock.hcl) per input; confirm c/nc config present

chunyiw8080 committed:
update name attribute values and docs for memcache (#159) (#194)

* update name attribute values and docs for memcache

* remove empty array terraform nc example

---------

Co-authored-by: chunyi wang <whunyi@foxmail.com>

Monneh committed:
dataform policies  (#197)

* feat: add dataform policies and inputs from gcp/service/dataforms branch

* feat: Add comprehensive Dataform service with 4 resources and documentation

- Add 4 Dataform resource types with complete policy coverage:
  - google_dataform_repository (8 policies)
  - google_dataform_repository_iam (1 policy)
  - google_dataform_repository_release_config (1 policy)
  - google_dataform_repository_workflow_config (1 policy)

- Add comprehensive security policies covering:
  - Deletion policy enforcement (FORCE vs DELETE)
  - Region allowlist restrictions
  - CMEK encryption requirements
  - Git authentication security (HTTPS/SSH)
  - Required security labels
  - IAM public access prevention
  - Cron schedule requirements
  - Service account enforcement

- Add complete documentation with security impact analysis:
  - 4 JSON resource documentation files with security rationales
  - 4 generated Markdown documentation files
  - Compliant/non-compliant examples for all security-critical arguments

- Add comprehensive test inputs:
  - Compliant (c.tf) and non-compliant (nc.tf) examples
  - Updated plan.json files for policy testing
  - Complete variable and configuration files

All policies tested and validated with OPA evaluation.

s223880826 committed:
Latest updated changes

s223880826 committed:
Delete inputs/gcp/beyondcorp/security_gateway_application_iam_binding/public_access_prevention directory

s223880826 committed:
Delete inputs/gcp/beyondcorp/security_gateway_application_iam_member/public_access_prevention directory

s223880826 committed:
Delete inputs/gcp/beyondcorp/security_gateway_iam_binding/public_access_prevention directory

s223880826 committed:
Delete inputs/gcp/beyondcorp/security_gateway_iam_member/public_access_prevention directory

s223880826 committed:
Delete inputs/gcp/beyondcorp/google_beyondcorp_app_connector/service_account_whitelist directory

s223880826 committed:
Updating new policy here

s223880826 committed:
Accommodated all the changes

s223880826 committed:
Updating the new beyondcorp policies

s223880826 committed:
Accommodated the changes

sumedh004 commented on policies/gcp/beyondcorp/google_beyondcorp_security_gateway_application/upstreams_whitelist/policy.rego:
```diff
@@ -1,38 +1,38 @@
+        "projects/my-proj/global/networks/prod-vpc",
+        "projects/my-proj/global/networks/shared-services-vpc"
+      ],
+      "policy_type": "pattern whitelist"
```
wrong usage of pattern whitelist. Watch the demo video on how it is used. If not able then you can switch to just whitelist

<img width="1062" height="191" alt="Image" src="https://github.com/user-attachments/assets/01496324-9f3b-420b-8b80-445b63af7acd" />

s223880826 replied:
Sure

sumedh004 reviewed (changes requested):
Just one correction in one policy needed. Rest is good

I dont see the documentation done, that needs to be done as well before merging

Also there are few merge conflicts, make sure those are addressed as well, otherwise we cant merge your branch

Saifmalikk committed:
Replaced SCC folders with version from gcp/service/sccv2 (#202)

christoraju085 committed:
Google BeyondCorp Latest

christoraju085 committed:
Google BeyondCorp Latest

s223880826 committed:
Updated Beyondcorp policies

Pat-Stuart committed:
Merge branch 'dev' into gcp/service/google_beyondcorp

s223880826 committed:
Updated the changes for upstream

s223880826 committed:
Updated the changes for upstream

sumedh004 reviewed (approved):
LGTM
